src/utils/package-json.ts

```diff
@@ -2,6 +2,7 @@ export type PackageDotJson = {
   scripts?: Record<string, string | undefined>;
   dependencies?: Record<string, string>;
   devDependencies?: Record<string, string>;
+  main?: string;
```

Comment:
NIT: could we rename `main` to `mainPath` or `mainSourceFilePath`? just to be more clear that the content of main will be the path to the main source file.

Reply:
I can add JSDoc, but the type represents the structure of `package.json` so we can't choose the keys.